### PR TITLE
Bux/fix spring oauth deps

### PR DESCRIPTION
**Pull request by sgandon**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
dependencies are messed up because of un-useful spring dependencies

**What is the new behavior?**
removed spring *compilation* scope deepencies and kept only *test* scope dependencies.
Also fixed some wrong dependencies along.

**Does this PR introduce a breaking change?**
- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
This may break the tests of certain components
pom needs to be adjusted accordingly.

WARNING : please merge after daikon PR merge : https://github.com/Talend/daikon/pull/93
Also when this PR is merged please merge this PR : https://github.com/Talend/components-ee/pull/30

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/773/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/773/)

File: components-common-oauth/pom.xml [hunk 1]
```diff
@@ -25,22 +25,6 @@
 			<version>8.1.17.v20150415</version>
```
Comment (gonchar-ivan) [inline components-common-oauth/pom.xml]: I think we will need to switch to older jetty version here.
Could be done later. First we need to see results of this PR

File: components-dataprep/pom.xml [hunk 2]
```diff
@@ -71,7 +71,7 @@
 		<dependency>
 			<groupId>org.apache.httpcomponents</groupId>
 			<artifactId>fluent-hc</artifactId>
-			<version>4.2.6</version>
+			<version>4.5.2</version>
```
Comment (gonchar-ivan) [inline components-dataprep/pom.xml]: Seems, we'll need to include it in Studio

Comment (sgandon) [inline components-dataprep/pom.xml]: nope it is embedded in the bundle for now.

**Pull request merged**